src/cryptonote_core/blockchain.cpp

```diff
@@ -3759,7 +3759,7 @@ bool Blockchain::check_tx_inputs(transaction& tx, tx_verification_context &tvc,
 
     if(have_tx_keyimg_as_spent(in_to_key.k_image))
     {
-      MERROR_VER("Key image already spent in blockchain: " << epee::string_tools::pod_to_hex(in_to_key.k_image));
+      MERROR_VER("Key image already spent in blockchain: " << epee::string_tools::pod_to_hex(in_to_key.k_image) << " of tx " << get_transaction_hash(tx));
```

Comment:
```suggestion
      MERROR_VER("Key image " << epee::string_tools::pod_to_hex(in_to_key.k_image) <<  " already spent in transaction " << get_transaction_hash(tx));
```

I think that this is a bit easier to read / understand.

Reply:
This tx hash is the tx hash of this tx the daemon is failing to verify, not the one that's in the chain already though

Reply:
Oh you're right. How about this then:

```suggestion
      MERROR_VER("Transaction " << get_transaction_hash(tx) << "spends key image already spent in blockchain: " << epee::string_tools::pod_to_hex(in_to_key.k_image));
```

Reply:
key image comes last in the line but still gives info on the bad tx